shared-utils/common.sh

```diff
@@ -173,8 +173,18 @@ function generate_rsa_edgecluster() {
     export RSA_PUB_FILE="${EDGE_SAFE_FOLDER}/${edgecluster}-rsa.key.pub"
 
     if [[ ! -f ${RSA_KEY_FILE} ]]; then
-        echo "RSA Key for Edge-cluster Cluster ${edgecluster} Not Found, creating one in ${EDGE_SAFE_FOLDER} folder"
-        ssh-keygen -b 4096 -t rsa -f ${RSA_KEY_FILE} -q -N ""
+        echo "RSA Key for Edge-cluster Cluster ${edgecluster} Not Found. Will create one in ${EDGE_SAFE_FOLDER} folder"
+
+        oc --kubeconfig=${KUBECONFIG_HUB} get -n ${cluster} secret/${cluster}-keypair >/dev/null 2>&1
```

Comment:
Good!

One thing to note is that the edgecluster deploy step will still "create" the secrets on every run. I think it's fine because it should result in a no-op